src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2446,71 +2452,91 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
             {
-                charsWritten = ConvertToBase64Array(outChars, inData, 0, bytes.Length, insertLineBreaks);
+                charsWritten = ConvertToBase64Array(outChars, inData, bytes.Length, insertLineBreaks);
                 return true;
             }
         }
 
-        private static unsafe int ConvertToBase64Array(char* outChars, byte* inData, int offset, int length, bool insertLineBreaks)
+        private static int[] CreateBase64PairsTable()
+        {
+            var table = new int[64 * 64];
+            for (int i = table.Length; i-- > 0;)
+            {
+                var firstChar = base64Table[i >> 6];
+                var secondChar = base64Table[i & 0x3f];
+                table[i] = BitConverter.IsLittleEndian
+                        ? firstChar | (secondChar << 16)
+                        : secondChar | (firstChar << 16);
+            }
+            return table;
+        }
+
+        private static unsafe int ConvertToBase64Array(char* outChars, byte* inData, int length, bool insertLineBreaks)
         {
+            int* outPairs = (int*)outChars;
             int lengthmod3 = length % 3;
-            int calcLength = offset + (length - lengthmod3);
-            int j = 0;
-            int charcount = 0;
-            // Convert three bytes at a time to base64 notation.  This will consume 4 chars.
-            int i;
-
-            // get a pointer to the base64Table to avoid unnecessary range checking
-            fixed (char* base64 = &base64Table[0])
+            int calcLength = length - lengthmod3;
+            byte a, b, c;
+
+            base64PairsTable ??= CreateBase64PairsTable();
+
+            // get a pointer to the base64PairsTable to avoid unnecessary range checking
+            fixed (int* base64Pairs = &base64PairsTable[0])
             {
-                for (i = offset; i < calcLength; i += 3)
+                while (calcLength > 0)
                 {
-                    if (insertLineBreaks)
+                    var rounds = insertLineBreaks
+                        ? Math.Min(calcLength, base64LineBreakPosition / 4 * 3)
+                        : calcLength;
+
+                    calcLength -= rounds;
+                    // Convert three bytes at a time to base64 notation.  This will consume 4 chars.
+                    for (; rounds > 0; rounds -= 3)
                     {
-                        if (charcount == base64LineBreakPosition)
-                        {
-                            outChars[j++] = '\r';
-                            outChars[j++] = '\n';
-                            charcount = 0;
-                        }
-                        charcount += 4;
+                        a = *inData++;
+                        b = *inData++;
+                        c = *inData++;
+                        *outPairs++ = base64Pairs[(a << 4) | (b >> 4)];
+                        *outPairs++ = base64Pairs[((b << 8) | c) & 0xfff];
                     }
```

Comment:
Maybe a manual loop unrolling would bring even more performance here.

Reply:
Vector operations based on AVX can give more (e.g. see https://lemire.me/blog/2018/01/17/ridiculously-fast-base64-encoding-and-decoding/)